### DISCUSSION TITLE
VPS搭建pandoraAI和node-chatgpt问题

### DISCUSSION THREAD

**Discussion by huoshenw** [viopsa233/AI-testers-Forum#95] [Q&A] [answered]
pandoraAI和node-chatgpt最新版搭建在同一个服务器上 端口都修改好了
为什么访问pandoraAI提问一点反应也没有 也没有报错 感觉是问题没有到node上面
pandoraAI也不知道怎么看日志 求大佬指点

**Comment by SoraRoseous** [comment]
浏览器f12看看请求的响应是什么，我估计多半是接口地址没给对

**Reply by huoshenw** [reply]
![image](https://github.com/viopsa233/AI-testers-Forum/assets/26380831/300cf882-3a21-4bb0-9c85-6106fa16697f)
请问y要查看哪一个的响应呢 小白不太懂@@

**Reply by SoraRoseous** [reply]
![image](https://github.com/viopsa233/AI-testers-Forum/assets/54582495/aa24335a-3ec4-4db6-9651-1beaf2ee208f)
点这个然后发一条消息，多出来的就是

**Comment by huoshenw** [comment]
> ![image](https://user-images.githubusercontent.com/54582495/255621598-aa24335a-3ec4-4db6-9651-1beaf2ee208f.png) 点这个然后发一条消息，多出来的就是
![image](https://github.com/viopsa233/AI-testers-Forum/assets/26380831/59805cf1-b5e3-4773-b6f8-ebc6f3005e9f)
![image](https://github.com/viopsa233/AI-testers-Forum/assets/26380831/012a8193-2681-4503-8737-1a63acc6ca09)

**Reply by SoraRoseous** [reply]
你看看这个请求的地址是你node里设置的地址和端口不

**Comment by huoshenw** [comment]
node的setting.js端口设置为3002
pandoraAI的env设置为http://localhost:3002

**Reply by SoraRoseous** [reply]
好像没啥问题，再看看防火墙呢

**Reply by huoshenw** [reply]
是看防火墙有没有放行3002是吗？还是要怎么操作。。。/(ㄒoㄒ)/~~

**Reply by SoraRoseous** [reply]
嗯

**Comment by huoshenw** [comment]
> 好像没啥问题，再看看防火墙呢

-A INPUT -p tcp -m tcp --dport 3000 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 3001 -j ACCEPT
-A INPUT -p tcp -m tcp --dport 3002 -j ACCEPT
我把这些全部放行了还是不行

**Reply by SoraRoseous** [reply]
看看conversation那个请求的状态码是什么

**Comment by huoshenw** [comment]
> 好像没啥问题，再看看防火墙呢

 lsof -i:3001
COMMAND    PID USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
node    469620 root   21u  IPv6 1618535      0t0  TCP *:3001 (LISTEN)
root@debian:~# lsof -i:3002
COMMAND    PID USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
node    437654 root   21u  IPv6 1554935      0t0  TCP localhost:3002 (LISTEN)
node    437654 root   22u  IPv4 1554936      0t0  TCP localhost:3002 (LISTEN)

**Comment by huoshenw** [comment]
> 看看conversation那个请求的状态码是什么

![image](https://github.com/viopsa233/AI-testers-Forum/assets/26380831/b509a697-b7cb-47f5-b8d6-41ed51023ee6)

**Comment by SoraRoseous** [comment]
等等，你这不是linux命令吗。。我突然意识到一个严重的问题，你是把PandoraAI放到了你的服务器上然后用你自己的电脑在访问是吗

**Comment by huoshenw** [comment]
> 等等，你这不是linux命令吗。。我突然意识到一个严重的问题，你是把PandoraAI放到了你的服务器上然后用你自己的电脑在访问是吗

是啊 我把pandoraAI和node搭在同一个服务器上 自己在用电脑网页访问

**Reply by SoraRoseous** [reply]
把localhost换成你服务器的ip地址

**Reply by huoshenw** [reply]
我也试过了 一样不行的

**Reply by SoraRoseous** [reply]
你在服务器上放的node，用localhost肯定不对，用ip addr里的地址才行

**Comment by huoshenw** [comment]
> 把localhost换成你服务器的ip地址



> 你在服务器上放的node，用localhost肯定不对，用ip addr里的地址才行

嗯嗯我也试过用http://ip:3002 还是不行

**Reply by SoraRoseous** [reply]
防火墙关了也不行吗？那我就不清楚了，当时我就踩过这些坑了

**Comment by huoshenw** [comment]
> 防火墙关了也不行吗？那我就不清楚了，当时我就踩过这些坑了

防火墙关了？是用什么命令

**Reply by SoraRoseous** [reply]
就是把那两个端口开开呀，你刚刚不是开了嘛，不过我用的是ufw，如果真是被防火墙墙了http的状态码应该是502吧好像，也不一定，不过我记得我当时好像是这个来着

**Reply by huoshenw** [reply]
apiOptions: {
        port: process.env.API_PORT || 3002,
        host: process.env.API_HOST || 'localhost',
setting.js这个部分的localhost需要改吗

**Reply by SoraRoseous** [reply]
要

**Reply by huoshenw** [reply]
Error: getaddrinfo ENOTFOUND http://129.150.x.12
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'http://129.150.x.12'
这里我改了就这样报错 不管是改成‘IP’还是‘http://ip’

**Reply by SoraRoseous** [reply]
是启动的时候报错吗还是聊天的时候报错啊

**Reply by huoshenw** [reply]
启动node报错

**Reply by SoraRoseous** [reply]
不加http也是一样的报错？还是不一样？

**Reply by huoshenw** [reply]
不加http是这个错误 
npm start

> @waylaidwanderer/chatgpt-api@1.37.2 start
> node bin/server.js

Error: listen EADDRNOTAVAIL: address not available 129.150.x.12:3002
    at Server.setupListenHandle [as _listen2] (node:net:1734:21)
    at listenInCluster (node:net:1799:12)
    at doListen (node:net:1948:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  code: 'EADDRNOTAVAIL',
  errno: -99,
  syscall: 'listen',
  address: '129.150.x.12',
  port: 3002
}

**Reply by SoraRoseous** [reply]
不要加端口号

**Comment by huoshenw** [comment]
> 就是把那两个端口开开呀，你刚刚不是开了嘛，不过我用的是ufw，如果真是被防火墙墙了http的状态码应该是502吧好像，也不一定，不过我记得我当时好像是这个来着



> 要
这里我改了就报错 不管是改成‘IP’还是‘http://ip’ 


> 要



> 是启动的时候报错吗还是聊天的时候报错啊



> 不加http也是一样的报错？还是不一样？
没有加啊
![image](https://github.com/viopsa233/AI-testers-Forum/assets/26380831/6f960c95-ee18-4c79-822c-838d26d45ea7)

**Reply by SoraRoseous** [reply, marked as answer]
那你改成0.0.0.0或者[::]试试
